File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/network/BiblioUlbraApi.java
package com.hgianastasio.biblioulbrav2.data.network;

import com.hgianastasio.biblioulbrav2.data.models.historybooks.HistoryBookEntity;
import com.hgianastasio.biblioulbrav2.data.models.loanbooks.LoanBookEntity;
import com.hgianastasio.biblioulbrav2.data.models.search.searchresult.SearchResultEntity;
import com.hgianastasio.biblioulbrav2.data.models.user.renew.RenewLoansResponseEntity;
import com.hgianastasio.biblioulbrav2.data.models.user.UserEntity;

import java.util.List;

import javax.inject.Singleton;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * Created by heitorgianastasio on 4/25/17.
 */
@Singleton
public interface BiblioUlbraApi {

    @GET("?funcao=h")
    Call<List<HistoryBookEntity>> getHistoryFromUser(@Query("cgu") String cgu);

    @GET("?funcao=e")
    Call<List<LoanBookEntity>> getLoansFromUser(@Query("cgu") String cgu);

    @GET("?funcao=l")
    Call<UserEntity> login(@Query("cgu") String cgu);

    @GET("?funcao=r")
    Call<RenewLoansResponseEntity> renew(@Query("cgu") String cgu);

    @GET("?funcao=b")
    Call<SearchResultEntity> search(
            @Query("busca") String search,
            @Query("cookie") String cookie,
            @Query("pag") int page,
            @Query("idioma") String language,
            @Query("mater") String media,
            @Query("campo") String field,
            @Query("local") String base
    );

}

File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/network/BiblioUlbraApi.kt
package com.hgianastasio.biblioulbrav2.data.network

import com.hgianastasio.biblioulbrav2.data.models.historybooks.HistoryBookEntity
import com.hgianastasio.biblioulbrav2.data.models.loanbooks.LoanBookEntity
import com.hgianastasio.biblioulbrav2.data.models.search.searchresult.SearchResultEntity
import com.hgianastasio.biblioulbrav2.data.models.user.UserEntity
import com.hgianastasio.biblioulbrav2.data.models.user.renew.RenewLoansResponseEntity
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query
import javax.inject.Singleton

/**
 * Created by heitorgianastasio on 4/25/17.
 */
@Singleton
interface BiblioUlbraApi {
    @GET("?funcao=h")
    fun getHistoryFromUser(@Query("cgu") cgu: String?): Call<List<HistoryBookEntity>>

    @GET("?funcao=e")
    fun getLoansFromUser(@Query("cgu") cgu: String?): Call<List<LoanBookEntity>>

    @GET("?funcao=l")
    fun login(@Query("cgu") cgu: String?): Call<UserEntity>

    @GET("?funcao=r")
    fun renew(@Query("cgu") cgu: String?): Call<RenewLoansResponseEntity>

    @GET("?funcao=b")
    fun search(
            @Query("busca") search: String?,
            @Query("cookie") cookie: String?,
            @Query("pag") page: Int,
            @Query("idioma") language: String?,
            @Query("mater") media: String?,
            @Query("campo") field: String?,
            @Query("local") base: String?
    ): Call<SearchResultEntity>
}